src/components/SigningModal/SigningModal.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect, useRef } from 'react'
+import PropTypes from 'prop-types'
+import { AppType } from '../../prop-types'
+import SignMessageFlow from './SignMessageFlow'
+import SignTransactionFlow from './SignTransactionFlow/SignTransactionFlow'
+
+function SigningModal({ apps, transactionBag, signatureBag, web3 }) {
+  const [modalVisible, setModalVisible] = useState(false)
+  const [isTransaction, setIsTransaction] = useState(false)
+
+  const prevTxBag = useRef()
+  const prevSigBag = useRef()
+
+  const handleClose = () => setModalVisible(false)
+
+  const sigBagUpdate =
+    signatureBag && signatureBag !== prevSigBag.current && signatureBag
+  const txBagUpdate =
+    transactionBag && transactionBag !== prevTxBag.current && transactionBag
+
+  useEffect(() => {
+    if (txBagUpdate) {
+      setModalVisible(true)
+      setIsTransaction(true)
+      prevTxBag.current = txBagUpdate
+    }
+  }, [txBagUpdate])
+
+  useEffect(() => {
+    if (sigBagUpdate) {
+      setModalVisible(true)
+      setIsTransaction(false)
+      prevSigBag.current = sigBagUpdate
+    }
+  }, [sigBagUpdate])
```

Comment:
I don't like this at all.. it's essentially the functional component equivalent of `componentDidUpdate`, I guess when the async api is available this can all go away.